File: web_ui/app/client_models/restaurant.ts
interface Customizations {
  [key: string]: {
    [key: string]: number;
  };
}

interface MenuItem {
  menu_item_id: string;
  name: string;
  price: number;
  description: string;
  customizations: Customizations;
}

interface MenuCategory {
  id: number; // Unique identifier for the category
  name: string; // Name of the menu category (e.g., "Appetizers", "Entrees")
  restaurantId: number; // ID of the restaurant to which the category belongs
}

interface Restaurant {
  created_at: string;
  email: string;
  id: number;
  location: string;
  name: string;
  phone_number: string;
  website: string;
}

export type { Restaurant, MenuCategory, MenuItem, Customizations };

// mealsync/12324/5/websocket_hash
// mealsync/[restaurant_id]/[table_id]

File: backend_server/models/restaurant.go
package models

import "time"

type Restaurant struct {
	ID          int       `json:"id"`
	Name        string    `json:"name" validate:"required"`        // Restaurant name is required
	Location    string    `json:"location"`                        // Location is optional
	PhoneNumber string    `json:"phone_number"`                    // Phone number is optional
	Email       string    `json:"email" validate:"required,email"` // Email should be required and valid
	Website     string    `json:"website"`                         // Website is optional
	OwnerID     int       `json:"ownerID" validate:"required"`     // OwnerID is required
	CreatedAt   time.Time `json:"created_at"`
}
